pkg/catalog/traffictarget_test.go

```diff
@@ -517,9 +517,9 @@ func TestListInboundTrafficTargetsWithRoutes(t *testing.T) {
 			expectedTrafficTargets: []trafficpolicy.TrafficTargetWithRoutes{
 				{
 					Name:        "ns-1/test-1",
-					Destination: identity.ServiceIdentity("sa-1.ns-1.cluster.local"),
+					Destination: identity.NewFromServiceAccountString("sa-1.ns-1.cluster.local"),
```

Comment:
This seems pretty misleading to me, that the `ServiceAccountString` has a trust domain. Before proceeding with this, I believe we should not mix the concept of `ServiceIdentity` with `K8sServiceAccount`, and that the type `ServiceIdentity` should be used as before in this file and other similar usages.

Reply:
@shashankram I don't quite understand what you mean by "`ServiceAccountString` has a trust domain".

This is a draft PR.   It is a struggle to get this to a small atomic PR that makes sense.  Also my thinking about what this `ServiceIdentity` keeps evolving as I read more code.  We might need to at some point step back and reasses.

Reply:
@draychev My comment is better summarized in https://github.com/openservicemesh/osm/issues/2218#issuecomment-819784148

When I see files such as `rbac_test.go` being changed to be ServiceAccount aware via the usage of `NewFromServiceAccountString()`, it seems to me this is not in alignment with #2218, where the intention is to use the `identity.ServiceIdentity` abstraction.